### DISCUSSION TITLE
Not purging; tailwind.config doesnt know the NODE_ENV

### DISCUSSION THREAD

**Discussion by getsetbro** [tailwindlabs/tailwindcss#2753] [Help] [answered]
Here is my build scripts - started with create-react-app:
```
"scripts": {
  "build": "npm run build:css && react-scripts build",
  "build:css": "postcss src/styles/tailwind.src.css -o src/styles/tailwind.css"
}
```

Here is my tailwind config:
```
console.log('env', process.env.NODE_ENV)
module.exports = {
  future: { purgeLayersByDefault: true, removeDeprecatedGapUtilities: true },
  purge: {content: ["./src/**/*.tsx", "./public/**/*.html"] },
  theme: { ...
```

When I `npm run build` tailwind does not purge because it doesnt know the mode is production. If I set purge to enabled it purges.

postcss config

```
// postcss.config.js
module.exports = {
    plugins: [
      require('postcss-import'),
      require('tailwindcss'),
      require('autoprefixer'),
    ]
  }
```

    "postcss": "^8.1.7",
    "postcss-cli": "^8.2.0",
    "postcss-import": "^13.0.0",
    "tailwindcss": "^1.9.6"
    "react-scripts": "^4.0.0",
    "autoprefixer": "^10.0.2",

**Comment by simonswiss** [comment, marked as answer]
Hey!

In your setup, Tailwind is being processed by `postcss` and not by the Create React App build script. Nothing tells your `build:css` script that NODE_ENV should be set to production.

You can either add `NODE_ENV=production` to that `build:css` script, or create another script specifically for prod build.

Alternatively, you can use [CRACO](https://www.npmjs.com/package/@craco/craco) to better integrate Tailwind with Create React App. It will let you configure PostCSS via webpack without ejecting.

We've got a setup guide coming out soon 👍

**Reply by getsetbro** [reply]
This works on mac but not windows
```
"build:css": "NODE_ENV=production postcss src/styles/tailwind.src.css -o src/styles/tailwind.css",
```

I guess I need the cross-env for that.

**Reply by OskarKlintrot** [reply]
Won't this work?

```
postcss --env production src/styles/tailwind.src.css -o src/styles/tailwind.css
```

**Comment by Mosquite12** [comment]
Just what I needed, worked a treat, thank you! [Publix Passport](https://www.publixoasis.us/)
